.internal-ci/charts/reserve-auditor/templates/api-statefulset.yaml

```diff
@@ -0,0 +1,174 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "reserve-auditor.fullname" . }}-api
+  labels:
+    {{- include "reserve-auditor.labels" . | nindent 4 }}
+    app.kubernetes.io/component: api
+spec:
+  replicas: {{ .Values.api.replicas }}
+  selector:
+    matchLabels:
+      {{- include "reserve-auditor.selectorLabels" . | nindent 6 }}
+      app.kubernetes.io/component: api
+  serviceName: {{ include "reserve-auditor.fullname" . }}-api
+  template:
+    metadata:
+      annotations:
+        {{- toYaml .Values.api.podAnnotations | nindent 8 }}
+      labels:
+        {{- include "reserve-auditor.selectorLabels" . | nindent 8 }}
+        app.kubernetes.io/component: api
+    spec:
+      imagePullSecrets:
+        {{- toYaml .Values.imagePullSecrets | nindent 8 }}
+      securityContext: {}
+      volumes:
+      - name: gnosis-config
+        configMap:
+          name: {{ include "reserve-auditor.fullname" . }}-gnosis
+      {{- if not .Values.persistence.enabled }}     
+      - name: data
+        emptyDir: {}
+      {{- end }}
+      containers:
+      {{- with .Values.mobilecoind }}
+      - name: mobilecoind
+        securityContext: {}
+        image: "{{ .image.repository }}:{{ .image.tag | default $.Chart.AppVersion }}"
+        imagePullPolicy: Always
+        ports:
+        - name: mobilecoind-rpc
+          containerPort: 3229
+        livenessProbe:
+          exec:
+            command:
+            - "/usr/local/bin/grpc_health_probe"
+            - "-addr=:3229"
+          failureThreshold: 3
+          periodSeconds: 30
+        readinessProbe:
+          exec:
+            command:
+            - "/usr/local/bin/grpc_health_probe"
+            - "-addr=:3229"
+          failureThreshold: 4
+          periodSeconds: 30
+        volumeMounts:
+        - name: data
+          mountPath: /ledger/
+        - name: data
+          mountPath: /mobilecoind/
+        env:
+        - name: "RUST_BACKTRACE"
+          value: "full"
+        - name: "RUST_LOG"
+          value: "info,rustls=warn,hyper=warn,tokio_reactor=warn,mio=warn,want=warn,reqwest=warn,rusoto_core=error,rusoto_signature=error,h2=error"
+        command:
+        - /bin/bash
+        - -ec
+        - |
+           /usr/bin/mobilecoind \
+             --chain-id {{ .chainID }} \
+             {{- range .peers }}
+             --peer {{ .url }} \
+             --tx-source-url {{ .txSource }} \
+             {{- end }}
+             --poll-interval {{ .pollInterval }} \
+             --quorum-set '{"threshold": {{ .threshold }}, "members": [
+               {{- range (initial .peers) -}}
+                 {{ include "reserve-auditor.quorumItem" . }},
+               {{- end -}}
+               {{- include "reserve-auditor.quorumItem" (last .peers) }}]}' \
```

Comment:
Follow-up for future rev:  I think @jgreat wrote a helper function  that renders this really well and would make it more readable.